### ISSUE TITLE
Code block without explicit language rendered as normal text

### ISSUE THREAD

**Issue by Abrynos** [wbaldoumas/markdown-colorcode#106]
## Description

Code blocks, which do not have a language declared at all will render as normal text.

e.g. take following markdown input:
![image](https://github.com/wbaldoumas/markdown-colorcode/assets/6608231/84344909-037c-4ebe-892e-01acbe656a9b)

## Expected Behaviour

The code is rendered as such without any syntax highlighting done.

The expected generated code:

```html
<p>Just some text</p>
<div style="color:#DADADA;background-color:#1E1E1E;"><pre>We put random stuff in a code-block without a language.
This should be rendered as a code-block.
</pre>
</div>
<p>Some more text</p>
```

[Screenshot](https://github.com/wbaldoumas/markdown-colorcode/assets/6608231/9d4bbd28-0073-4e69-8e40-a391ab90ccef)

## Actual Behaviour

The code is rendered as normal text within the html.

The actually generated code:
```html
<p>Just some text</p>
<pre><code>We put random stuff in a code-block without a language.
This should be rendered as a code-block.
</code></pre>
<p>Some more text</p>
```

[Screenshot](https://github.com/wbaldoumas/markdown-colorcode/assets/6608231/7f6e9ba8-b342-48b3-8b3b-9c5ae58769a9)

## Affected Version

`1.1.1` - Possibly prior versions; I did not test this.

## Steps to Reproduce

1. Save example markdown I provided above
2. Transform to html using the library
3. look at the result

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/wbaldoumas/markdown-colorcode/blob/main/CONTRIBUTING.md)
- [x] I have verified this does not duplicate an existing issue

**Comment by wbaldoumas** [comment]
@Abrynos thanks for your contribution by opening this issue and associated pull request! 😄

Could you expand on the benefits of this proposed change? I tossed the HTML generated by the current behavior (which leverages the `<code>` tag generated by Markdig under the hood) and the HTML generated by your suggested behavior (using `style) into a previewer and the end result is basically identical.

Using `<code>`:

![image](https://github.com/wbaldoumas/markdown-colorcode/assets/45316999/10dbc1e3-1449-4e75-af90-4f779e535071)


Using `style`:

![image](https://github.com/wbaldoumas/markdown-colorcode/assets/45316999/e3f9ab26-d725-4c97-a085-aea09b551c3a)

**Comment by Abrynos** [comment]
> Could you expand on the benefits of this proposed change?

Of course. The issue mainly comes into play when using bootstrap. Here's a full reproduction example:

~~~csharp
using System.IO;
using System.Threading.Tasks;
using Markdig;
using Markdown.ColorCode;

namespace MdColorCodeReproduction;

public static class Program
{
    private const string MarkdownInput = @"
# Test

```text
This is declared as a code-block and should be rendered as such.
```

";

    private const string HtmlHeader = @"
<!DOCTYPE html>
<html lang=""en"" data-bs-theme=""dark"">
  <head>
    <meta charset=""utf-8""/>
    <meta http-equiv=""content-type"" content=""text/html""/>
    <meta http-equiv=""Cache-Control"" content=""no-cache, no-store, must-revalidate""/>
    <meta http-equiv=""Pragma"" content=""no-cache""/>
    <meta http-equiv=""Expires"" content=""0""/>
    <link href=""https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/dist/css/bootstrap.min.css""
          rel=""stylesheet""
          integrity=""sha384-4bw+/aepP/YC94hEpVNVgiZdgIC5+VKNBQNGCHeKRQN+PtmoHDEXuppvnDJzQIu9""
          crossorigin=""anonymous""/>
  </head>
  <body class=""container mb-2"">
";

    private const string HtmlFooter = @"
    <script src=""https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/dist/js/bootstrap.bundle.min.js""
            integrity=""sha384-HwwvtgBNo3bZJJLYd8oVXjrBZt8cqVSpeBNS5n7C8IVInixGAoxmnlMuBnhbgrkm""
            crossorigin=""anonymous""/>
  </body>
</html>
";

    public static async Task<int> Main(string[] args)
    {
        MarkdownPipeline pipeline = new MarkdownPipelineBuilder()
            .UseBootstrap()
            .UseColorCode()
            .Build();

        string body = Markdig.Markdown.ToHtml(MarkdownInput, pipeline);
        await File.WriteAllTextAsync("./output.html", HtmlHeader + body + HtmlFooter).ConfigureAwait(false);

        return 0;
    }
}
~~~

Of course it does not matter whether `data-bs-theme="dark"` is set or not. I just copied that over from our actual code.

**Comment by wbaldoumas** [comment]
@Abrynos thanks for the full reproduction example!

I see that this is indeed an issue, but the issue lies in the fact that that `text` is not [one of the provided languages](https://github.com/CommunityToolkit/ColorCode-Universal/tree/main/ColorCode.Core/Compilation/Languages) within `ColorCode.Core`, not that `Markdown.ColorCode` isn't handling things correctly. The right place to open this issue and pull request to add your [TextLanguage]( https://github.com/wbaldoumas/markdown-colorcode/pull/107/files#diff-7b100a717950421f992b5e1cd000000d72bdf0bf7245045f1cb034538d0f7391R9) would be over in the [ColorCode-Universal]( https://github.com/CommunityToolkit/ColorCode-Universal) library, right alongside all of the other languages there. The benefit of doing this there is that others using that library would benefit as well. 😄 

Once that change is merged and a new version of `ColorCode.Core` is released, I'll bump the version in `Markdown.ColorCode` and everything will work as you expect it to.

**Comment by Abrynos** [comment]
@wbaldoumas The issue still remains if there is no language provided in the markdown at all. If someone explicitly declares something to be a code-block, it should be rendered as such even if we don't know how to colour-code it.

**Comment by wbaldoumas** [comment]
@Abrynos agreed! 

My stance on this, though, is still to add that `TextLanguage` you created to the main `ColorCode.Core` package. Once that's there we can fall back to it if no language is provided, similar to how you are doing that [here](https://github.com/wbaldoumas/markdown-colorcode/pull/107/files#diff-4b1c23612e429ca57001e2610d9c657e33f61f52931079f6a2fb361881aee674R71).

**Comment by Abrynos** [comment]
I've already opened a merge request for the main repository: CommunityToolkit/ColorCode-Universal#37. I'll continue working on the PR over here as soon as a decision is taken there.

**Comment by wbaldoumas** [comment]
@Abrynos give version `2.0.0-preview-1` of this package a try.

I've introduced functionality to add your own languages as well as set a default language ID in https://github.com/wbaldoumas/markdown-colorcode/pull/110. 

The default language ID will be used in the scenarios when either the language ID for the code block is not provided or when an unsupported language ID is used. This should give you the functionality that you're interested in without waiting for your PR to get merged into ColorCode-Universal (although I still think that PR should be merged).

Let me know what you think.

**Comment by Abrynos** [comment]
@wbaldoumas Thanks, but we're good for now. As a (hopefully temporary) fix we're already using `ColorCode.Core.Languages.Load(new OurLanguage())` with `OurLanguage` being quite a hack. Although the new API seems intriguing.

**Comment by wbaldoumas** [comment]
@Abrynos I've merged and released v2.0.0 of this library with the new, more flexible API. 

You should be able to use this to address the initial issue you posted about by providing your custom `PlainText` language and providing its ID as your desired fallback language for when nothing is specified. It's also a bit less opinionated than the initial suggested solution and allows other users to configure different fallback behavior, if desired.

I'm closing out this issue as addressed, but if you notice any problems, don't hesitate to open a new one. Thank you for your contributions! 🎉
